### PR TITLE
Update README and add hack script for demo setup

### PR DESCRIPTION
**Pull request by nrfox**
Update README with cypress requirements and add a hack script for installing demos necessary for cypress tests.

### CODE REVIEW

File: hack/istio/install-testing-demos.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+fi
+
+echo "CLIENT_EXE=${CLIENT_EXE}"
+echo "IS_OPENSHIFT=${IS_OPENSHIFT}"
+
+# Installed demos should be the exact same for both environments.
+# Only the args passed to the scripts differ from each other.
+if [[ "${IS_OPENSHIFT}" = "true" ]]; then
+  echo "Deploying bookinfo demo..."
+  /bin/bash "${SCRIPT_DIR}"/install-bookinfo-demo.sh
```
Comment (jmazzitelli) [inline hack/istio/install-testing-demos.sh:50]: a) do you need `/bin/bash` ? The .sh files have the `#!/bin/bash` headers that tell the OS what shell to use.
b) the quotes around `"${SCRIPT_DIR}"` I assume are to protect against spaces in the path? If so, the placing of the quotes looks weird to me. I would think it should just be `"${SCRIPT_DIR}/install-bookinfo-demo.sh"` (the entire string wrapped)

File: .github/workflows/kiali-ci.yml [hunk 2]
```diff
@@ -202,5 +202,5 @@ jobs:
         working-directory: tests/integration/tests
 
       - name: Get kiali pod logs when tests fail
-        if: ${{ steps.integration-tests.outcome == 'failure' }}
+        if: steps.integration-tests.outcome == 'failure'
         run: kubectl logs deployments/kiali -n istio-system
```
Comment (jmazzitelli) [inline .github/workflows/kiali-ci.yml:206]: this is weird, i thought we already changed this? Yesterday @leandroberetta had changed this, I thought? This is to ensure we get the logs regardless of the `deployment.instance_name` we may have tested with:
```suggestion
        run: kubectl logs -l app.kubernetes.io/name=kiali -n istio-system
```

Comment (leandroberetta) [reply .github/workflows/kiali-ci.yml:206]: @jmazzitelli my PR is not merged yet, changes are still there.

File: README.adoc [hunk 3]
```diff
@@ -246,6 +246,16 @@ cd $KIALI_SOURCES/kiali/hack
...
 
+Integration testing is done with cypress. More information can be found link:./frontend/cypress/README.md[here].
+
+Requirements:
+- Istio
+- Kiali
+- bookinfo demo app
+- error rates demo app
+
+You can use link:./hack/install-testing-demos.sh[this script] to install all the neccessary demo apps for testing. The script supports both openshift and non-openshift deployments.
```
Comment (jmazzitelli) [inline README.adoc:257]:
```suggestion
You can use link:./hack/istio/install-testing-demos.sh[this script] to install all the neccessary demo apps for testing. The script supports both openshift and non-openshift deployments.
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

**Review by jmazzitelli — APPROVED** [review]

**Review by jmazzitelli — APPROVED** [review]

**Comment by nrfox** [comment]
Some of the tests are starting to flake. My guess is it's related to the increased resource usage from deploying additional demo apps. The GH actions VMs only have 7 GB of mem and 2 CPU so they are pretty resource constrained. I'm going to try tweaking some of the resource requirements for various things to see if we can get it stable again.

**Comment by nrfox** [comment]
Adding "do not merge" label until test flakiness has been resolved. Will be pushing commits for testing CI changes.

**Review by jmazzitelli — APPROVED** [review]

**Review by jmazzitelli — APPROVED** [review]

**Comment by nrfox** [comment]
This is ready for :eyes: again @jmazzitelli. Turns out the flaky tests were from patternfly changing the dom id for a `Select` element when you don't pass one explicitly. Still I think it's worth keeping the resource adjustments for the proxies included in this PR to try and minimize the footprint.

**Pull request merged**